### DISCUSSION TITLE
A gui without using a browser + a suggestion

### DISCUSSION THREAD

**Discussion by Amin456789** [LostRuins/koboldcpp#452] [Ideas]
Hello

Thank you for making this

can we please have a gui that doesn't rely on browser? like gpt4all, this way all our ram will be for this ai, of course it will be optional so people can use webui or a local gui without any dependencies.

Also, could we please have an option to upload our image to the chat for immersion? it doesn't have to answer but with minigpt4.cpp integration ai can comment on the image, or using stable-diffusion.cpp to generate image in the middle of chat

kind regards

**Comment by LostRuins** [comment]
Hmm perhaps directly using llama.cpp would be more suitable. KoboldCpp is designed to work with a GUI.

**Comment by Amin456789** [comment]
Thank you for fast reply! 
I love KobodCpp and don't know anything about llama.cpp, I think it doesn't have every options such as adventure mode etc...
I edited my question for a suggestion, could u take a look please?

**Reply by LostRuins** [reply]
Yes you can upload your image! you just need to select "Aesthetic" chat UI and you click the gear icon to choose your own image. 

![image](https://github.com/LostRuins/koboldcpp/assets/39025047/0ae5617b-4bb9-47b1-9216-35f79f78efc6)

additionally you can also generate images via stable horde. just choose a model and it will add a button 

![image](https://github.com/LostRuins/koboldcpp/assets/39025047/bb3ad494-e6e1-49f6-8da5-9321be64dc43)

**Comment by Amin456789** [comment]
That's great! Thank you so much!
